sound/soc/sof/intel/hda-dai.c

```diff
@@ -27,20 +27,30 @@ static bool hda_use_tplg_nhlt;
 module_param_named(sof_use_tplg_nhlt, hda_use_tplg_nhlt, bool, 0444);
 MODULE_PARM_DESC(sof_use_tplg_nhlt, "SOF topology nhlt override");
 
+static struct snd_sof_dev *widget_to_sdev(struct snd_soc_dapm_widget *w)
+{
+	struct snd_sof_widget *swidget = w->dobj.private;
+	struct snd_soc_component *component;
+
+	if (!swidget)
+		return NULL;
```

Comment:
NULL can be returned but we don't check it.

Reply:
that's true. I wasn't sure what to do here, because defensive programming basics would be to avoid de-referencing a NULL pointer, but the rest of the code does not test this possibility. the 'sdev' pointer is taken as is. I didn't feel too eager to change everything for something that may never happen.

@ranj063 what would be your recommendation on deal with such error cases?

Reply:
@plbossart I dont think swidget can ever be NULL here because we bind the BE DAI to the playback/capture widget during topology parsing and if we couldn't, we'd have failed a lot earlier.

Reply:
ok, i'll add a comment along those lines here.